### PR TITLE
reactivate some tests ..

### PR DESCRIPTION
**Pull request by oberstet**

### CODE REVIEW

**Comment by oberstet** [comment]
it works locally

```
====================================================== 8 passed in 321.89 seconds =======================================================
2022-03-28T05:34:02+0200 Main loop terminated.
2022-03-28T05:34:02+0200 Unhandled error in Deferred:
2022-03-28T05:34:02+0200 
________________________________________________________________ summary ________________________________________________________________
  sphinx: commands succeeded
  flake8: commands succeeded
  mypy: commands succeeded
  yapf: commands succeeded
  bandit: commands succeeded
SKIPPED:  py37-pinned-trial: InterpreterNotFound: python3.7
SKIPPED:  py37-unpinned-trial: InterpreterNotFound: python3.7
SKIPPED:  py37-abtrunk-trial: InterpreterNotFound: python3.7
  py39-pinned-trial: commands succeeded
  py39-unpinned-trial: commands succeeded
  py39-abtrunk-trial: commands succeeded
SKIPPED:  py310-pinned-trial: InterpreterNotFound: python3.10
SKIPPED:  py310-unpinned-trial: InterpreterNotFound: python3.10
SKIPPED:  py310-abtrunk-trial: InterpreterNotFound: python3.10
SKIPPED:  pypy38-pinned-trial: InterpreterNotFound: pypy3.8
SKIPPED:  pypy38-unpinned-trial: InterpreterNotFound: pypy3.8
SKIPPED:  pypy38-abtrunk-trial: InterpreterNotFound: pypy3.8
SKIPPED:  py37-cli: InterpreterNotFound: python3.7
  py39-cli: commands succeeded
SKIPPED:  py310-cli: InterpreterNotFound: python3.10
SKIPPED:  pypy38-cli: InterpreterNotFound: pypy3.8
SKIPPED:  py37-examples: InterpreterNotFound: python3.7
  py39-examples: commands succeeded
SKIPPED:  py310-examples: InterpreterNotFound: python3.10
SKIPPED:  pypy38-examples: InterpreterNotFound: pypy3.8
  functests-cb: commands succeeded
  functests-cfc: commands succeeded
  congratulations :)
(cpy39_1) (base) oberstet@intel-nuci7:~/scm/crossbario/crossbar$ 
```

**Comment by oberstet** [comment]
```
===================================== 291 passed, 46 skipped, 3 xfailed, 1 warnings in 4.27 seconds =====================================
________________________________________________________________ summary ________________________________________________________________
  pytest: commands succeeded
  congratulations :)
```

**Comment by oberstet** [comment]
local test is now working for:

* [ ] sphinx
* [ ] flake8
* [ ] mypy
* [ ] yapf
* [ ] bandit
* [ ] pytest
* [ ] py37-pinned-trial
* [ ] py37-unpinned-trial
* [ ] py37-abtrunk-trial
* [ ] py39-pinned-trial
* [ ] py39-unpinned-trial
* [ ] py39-abtrunk-trial
* [ ] py310-pinned-trial
* [ ] py310-unpinned-trial
* [ ] py310-abtrunk-trial
* [ ] pypy38-pinned-trial
* [ ] pypy38-unpinned-trial
* [ ] pypy38-abtrunk-trial
* [ ] py37-examples
* [ ] py39-examples
* [ ] py310-examples
* [ ] pypy38-examples
* [ ] functests-cb
* [ ] functests-cfc
* [ ] py39-api-1
* [ ] pypy38-api-1
* [ ] py39-cli-0
* [ ] pypy38-cli-0
* [ ] py39-cli-1
* [ ] pypy38-cli-1
* [ ] py39-cli-2
* [ ] pypy38-cli-2
* [ ] py39-cli-3
* [ ] pypy38-cli-3
* [ ] py39-automate-1
* [ ] pypy38-automate-1
* [ ] py39-automate-2
* [ ] pypy38-automate-2
* [ ] py39-xbrnetwork-1
* [ ] pypy38-xbrnetwork-1

----

```
  sphinx: commands succeeded
  flake8: commands succeeded
  mypy: commands succeeded
  yapf: commands succeeded
  bandit: commands succeeded
  py39-pinned-trial: commands succeeded
  py39-unpinned-trial: commands succeeded
  py39-abtrunk-trial: commands succeeded
  py39-examples: commands succeeded
  pytest: commands succeeded
  functests-cb: commands succeeded
  functests-cfc: commands succeeded
  py39-api-1: commands succeeded
```

**Comment by oberstet** [comment]
it works locally via tox, but fails on GH actions:

```
==================================== ERRORS ====================================
_______ ERROR collecting test/functests/cfctests/test_cfc_rlink_mesh.py ________
test/functests/cfctests/test_cfc_rlink_mesh.py:40: in <module>
    node1_pubkey = _read_node_key(node1_cbdir, private=False)['hex']
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/crossbar/common/key.py:86: in _read_node_key
    raise Exception('no node key file found at {}'.format(node_key_path))
E   Exception: no node key file found at ./test/cf1/.crossbar/key.pub
_________ ERROR collecting test/functests/cfctests/test_cfc_rlinks.py __________
test/functests/cfctests/test_cfc_rlinks.py:32: in <module>
    node1_pubkey = _read_node_key(node1_cbdir, private=False)['hex']
/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/crossbar/common/key.py:86: in _read_node_key
    raise Exception('no node key file found at {}'.format(node_key_path))
E   Exception: no node key file found at ./test/cf1/.crossbar/key.pub
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 2 error in 0.38 seconds ============================
```

File: requirements-latest.txt [hunk 1]
```diff
@@ -72,7 +73,8 @@ validate-email>=1.3
 vmprof>=0.4.15
 watchdog>=0.8.3
 web3>=5.13.1
-werkzeug>=0.14.1
+# ImportError: cannot import name 'escape' from 'werkzeug.utils'
+werkzeug>=0.14.1,<2
```
Comment (Bastian-Krause) [inline requirements-latest.txt:77]: Why `<2`? `werkzeug.utils.escape` was removed in 2.1.0, see [changelog](https://werkzeug.palletsprojects.com/en/2.1.x/changes/#version-2-1-0).

Comment (oberstet) [reply requirements-latest.txt:77]: yeah, maybe <2.1 would work as well. not sure. eg flask needs werkzeug etc., but I didn't test it exhaustively to figure it out https://gist.github.com/oberstet/726787d3acb0da736ee194c0e3af914f

Comment (oberstet) [reply requirements-latest.txt:77]: also, this should be looked at together with https://github.com/crossbario/crossbar/blob/ecc8b4e0d9f8fa3efa7d59755cb2e16151055870/requirements-min.txt#L32 .. at least I ran into both at the same time recently ..

**Pull request merged**